course/auth.py

```diff
@@ -680,9 +774,35 @@ def user_profile(request):
 
     user_form = None
 
+    def has_verified_inst_id(user):
```

Comment:
My vote would be for making sure the meaning of all the code here is crystal-clear and completely obvious from its naming. This code is sensitive in the sense that if it mishandles student info, then that's either a [FERPA](https://en.wikipedia.org/wiki/Family_Educational_Rights_and_Privacy_Act) violation (on my end) or an opportunity for cheating by students, or both.